pkg/parca/parca.go

```diff
@@ -856,3 +887,51 @@ func getDiscoveryConfigs(cfgs []*config.ScrapeConfig) map[string]discovery.Confi
 	}
 	return c
 }
+
+func BucketURIFromConfig(bucketCfg []byte) (string, error) {
+	bucketConf := &client.BucketConfig{}
+	if err := yaml.Unmarshal(bucketCfg, bucketConf); err != nil {
+		return "", fmt.Errorf("failed to unmarshal bucket config: %w", err)
+	}
+
+	type Config struct {
+		Bucket string `yaml:"bucket"`
+	}
+
+	config, err := yaml.Marshal(bucketConf.Config)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal content of bucket configuration: %w", err)
+	}
+
+	switch strings.ToUpper(string(bucketConf.Type)) {
+	case string(client.GCS):
+		var cfg Config
+		if err := yaml.Unmarshal(config, &cfg); err != nil {
+			return "", err
+		}
+		return filepath.Join("gs://", cfg.Bucket, bucketConf.Prefix), nil
```

Comment:
how does authentication for these work?

Reply:
It's just using the thanos objstore library. So the same way that authentication currently works in Parca